alarm_control_panel-card/alarm_control_panel-card.js

```diff
@@ -29,14 +14,43 @@ class AlarmControlPanelCard extends HTMLElement {
   }
 
   set hass(hass) {
-    this.myhass = hass;
     const entity = hass.states[this._config.entity];
-    if (entity && entity.state != this._state) {
-      this._state = entity.state;
-      this._updateCardContent(entity);
+
+    if (entity) {
+      this.myhass = hass;
+      if(!this.shadowRoot.lastChild) {
+        this._createCard(entity);
```

Comment:
That is a simplistic example not reflecting actual complete lifecycle of a custom element. You can read more about it here: https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#Using_the_lifecycle_callbacks and here: https://stenciljs.com/docs/component-lifecycle

Example of different approach to lifecycle dom creation: https://github.com/custom-cards/vertical-stack-in-card/blob/master/vertical-stack-in-card.js . See `connectedCallback`

We can still use your method for now until I find a better way to represent the complete lifecycle.

Reply:
Thanks for the pointers! I read through all of them. The `connectedCallback` is interesting part of the lifecycle.

At this point, I'm going to leave the code as is. There are two challenges that make any changes interesting...

First, `config` is not defined until `setConfig` is called. `config` is needed to creating the style.

Second, `entity` is not available until `set hass` is called. That's required to decide if the `input` field and the keypad should be created.

Both of those can be overcome, and it may not be "pretty", but no sense refactoring code until the lifecycle is sorted out.

Reply:
Usually the approach would be to create a skeleton of the best guess and then just populate data in as you receive it. But I agree, it can stay as it is until next refactoring. 

One nice enhancement would be to remove the text field when you have "auto enter". As we know the length of the code on the UI we can use some round circles with some filling to represent keys entered. I've seen someone on the forums suggesting this and I think it's a nice feature to have.

Reply:
I like having the text field. Use it all the time when testing. On a phone/tablet I would likely not use it. So the option to hide/show it would still be appreciated.